### PR TITLE
Andyfuse work

### PR DESCRIPTION
**Pull request by dchaplinsky**

### CODE REVIEW

File: new_solution.py [hunk 1]
```diff
@@ -26,6 +74,52 @@ def _jsrch(self, path, doc=None):
...
+
+            if not totalArea:
+                return 0
+
+            if not areaUnits:
+                return float(totalArea)
+
+            return float(totalArea) * areas_koef.get(areaUnits, 1)
+        except ValueError:
+            return 0
```
Comment (dchaplinsky) [inline new_solution.py:93]: logging.debug(...)

File: new_solution.py [hunk 2]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                   "relations_other" in fm_member:
+                   extract[i]["subjectRelation"] += \
+                                    ", " + fm_member["relations_other"]
+        elif raw_family_info:
+            for i, fm_member in enumerate(self._parse_raw_family_string(
+                                                raw_family_info)):
+                extract[i] = {}
+                extract[i]["raw"] = fm_member.get("raw", "")
+                extract[i]["subjectRelation"] = fm_member.get("relations", "")
+                extract[i]["bio_declomua"] = fm_member.get("family_name", "")
```
Comment (dchaplinsky) [inline new_solution.py:226]: What is bio_declcomua?

File: new_solution.py [hunk 3]
```diff
@@ -118,22 +1011,26 @@ def convert(self):
...
 
+        new_file = os.path.join(out_dir, subdir, basename)
         with open(file_name, 'r') as infile:
             try:
                 old_json = json.load(infile)
             except json.decoder.JSONDecodeError:
                 logger.error('Empty or broken file: {}'.format(file_name))
                 continue
-
+
```
Comment (dchaplinsky) [inline new_solution.py:1033]: Hanging spaces :-\

File: new_solution.py [hunk 4]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                                 (None, 'otherObjectType', ""),
+                                 ('costs_rent', 'costRent_declcomua', ""),
+                                 (None, 'costAssessment_extendedstatus', ""),
+                                 (None, 'ua_housePartNum_extendedstatus', ""),
+                                 (None, 'ua_apartmentsNum_extendedstatus', "")
+                                ],
+                                estate
+                        )
+                        extract[record_counter]['objectType'] = \
+                            estate_desc_dict[estate_id] + ' (стара форма)'
```
Comment (dchaplinsky) [inline new_solution.py:289]: стара форма, what for?

File: new_solution.py [hunk 5]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                                 ('costs', 'costDate', ""),
+                                 (None, 'district', ""),
+                                 (None, 'postCode', ""),
+                                 (None, 'iteration', ""),
+                                 ('space', 'totalArea', ""),
+                                 (None, 'ua_street', ""),
+                                 (None, 'objectType', ""),
+                                 (None, 'owningDate', ""),
+                                 (None, 'objectType', ""),
+                                 (None, 'ua_cityType', ""),
```
Comment (dchaplinsky) [inline new_solution.py:276]: Please clarify it. In recent declarations I can see region set in ua_cityType field.

For example:
https://public-api.nazk.gov.ua/v1/declaration/43982943-477f-40ea-9dd2-57c834994819
https://declarations.com.ua/declaration/nacp_43982943-477f-40ea-9dd2-57c834994819

File: new_solution.py [hunk 6]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                                 (None, 'ua_postCode', ""),
+                                 ('address', 'ua_streetType', ""),
+                                 (None, 'costAssessment', ""),
+                                 (None, 'otherObjectType', ""),
+                                 ('costs_rent', 'costRent_declcomua', ""),
```
Comment (dchaplinsky) [inline new_solution.py:281]: In case the object is rented you should also set "ownershipType": "Оренда"

File: new_solution.py [hunk 7]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+            for vehicle_id, vehicle_items in vehicle_info.items():
+                owner_id = ('1'
+                            if int(vehicle_id) in range(35,40)
+                            else 'family')
+                for vehicle in vehicle_items:
+                    extract[record_counter] = \
+                        self._convert_using_rules(
+                            [
+                             ('brand', 'brand', ""),
+                             ('brand_info', 'brand_info_declcomua', ""),
```
Comment (dchaplinsky) [inline new_solution.py:366]: put that info into the model field. Not ideal fit, but a bit better

File: new_solution.py [hunk 8]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                             (None, 'person', ""),
+                             (None, 'model', ""),
+                             (None, 'rights', {}),
+                             ('sum', 'costDate', ""),
+                             ('sum_rent', 'costRent_declcomua', ""),
```
Comment (dchaplinsky) [inline new_solution.py:371]: Again, in the rights dict you should mention, if it's a rented vehicle.

File: new_solution.py [hunk 9]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                             (None, 'iteration', ""),
+                             (None, 'objectType', ""),
+                             (None, 'owningDate', ""),
+                             ('year', 'graduationYear', ""),
+                             (None, 'otherObjectType', "")
+                            ],
+                            vehicle
+                        )
+                    extract[record_counter]['objectType'] = \
+                            vehicle_desc_dict[vehicle_id] + ' (стара форма)'
```
Comment (dchaplinsky) [inline new_solution.py:381]: Why mention that this is old form?

File: new_solution.py [hunk 10]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+            "step_6": extract
+        }
+
+    def _convert_step7(self):
+        extract = {}
+        record_counter = 1
+        papers_desc = 'Номінальна вартість цінних паперів  (стара форма)'
+        papers_info = {}
+        temp_dict = self._jsrch('banks')
+        # self._jsrch('banks.47') - does not work - jmespath.exceptions.ParseError
```
Comment (dchaplinsky) [inline new_solution.py:432]: Hmm, ok

File: new_solution.py [hunk 11]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+        papers_info[47] = temp_dict.get('47', None) if temp_dict else {}
```
Comment (dchaplinsky) [inline new_solution.py:433]: No need to specify default value of None

File: new_solution.py [hunk 12]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+        papers_info[52] = temp_dict.get('52', None) if temp_dict else {}
+        outer_papers_info = temp_dict.get('48', None) if temp_dict else {}
+
+        if papers_info:
+            for paper_id, paper_items in papers_info.items():
+                owner_id = ('1'
+                            if paper_id == 47
+                            else 'family')
+                if paper_items:
```
Comment (dchaplinsky) [inline new_solution.py:442]: Probably, it's a bit better to use
```
if not paper_items:
  continue
```

will save you a bit of nesting

File: new_solution.py [hunk 13]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                    for paper in paper_items:
+                        if ('sum' in paper and paper['sum'] != "") or \
```
Comment (dchaplinsky) [inline new_solution.py:444]: if paper.get("sum") or paper....:

will simplify your life a little bit

File: new_solution.py [hunk 14]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                                 (None, 'rights', {}),
+                                 (None, 'emitent', ""),
+                                 (None, 'owningDate', ""),
+                                 (None, 'emitent_type', ""),
+                                 (None, 'typeProperty', papers_desc),
+                                 (None, 'otherObjectType', ""),
+                                 (None, 'subTypeProperty1', ""),
+                                 (None, 'subTypeProperty2', ""),
+                                 (None, 'emitent_ua_lastname', ""),
+                                 ('sum_foreign_comment', 'emitent_eng_fullname', ""),
```
Comment (dchaplinsky) [inline new_solution.py:462]: Are you sure those two fields are good match?

File: new_solution.py [hunk 15]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+                            record_counter += 1
+
+        return {    
+            "step_7": extract
+        }
+
+    def _convert_step8(self):
+        extract = {}
+        record_counter = 1
+        papers_desc = 'Розмір внесків до статутного капіталу товариства, підприємства, організації  (стара форма)'
```
Comment (dchaplinsky) [inline new_solution.py:531]: can we match this description and description above to the options, used in the current (new, nacp) form?

File: new_solution.py (hunk 10 above)
Comment (dchaplinsky) [reply new_solution.py:432]: Try banks."47"

File: new_solution.py [hunk 16]
```diff
@@ -70,25 +164,801 @@ def _convert_step1(self):
...
+        }
+
+    def _convert_step12(self):
+        extract = {}
+        record_counter = 1
+        papers_desc = 'Сума коштів на рахунках у банках та інших фінансових установах (стара форма)'
+        papers_info = {}
+        temp_dict = self._jsrch('banks')
+        # self._jsrch('banks.47') - does not work - jmespath.exceptions.ParseError
+        papers_info[45] = temp_dict.get('45', None) if temp_dict else {}
```
Comment (dchaplinsky) [inline new_solution.py:623]: I understand, that you've copypasted this step and step above from the step6.

But can you at least give those variables a meaningful names?

File: new_solution.py (hunk 2 above)
Comment (ghost) [reply new_solution.py:226]: "Если из декларации можно выделить фамилию, имя и отчество, они записываются отдельно в соответствующие поля. Если нет — они записываются в отдельное поле ""bio_declcomua"" (вместе с датой рождения) "

File: new_solution.py (hunk 10 above)
Comment (andyfuse) [inline new_solution.py]: >>> import jmespath
>>> a = {'banks': {47: 'bla'}}
>>> jmespath.search('banks."47"', a)
>>> jmespath.search('banks.47', a)
jmespath.exceptions.ParseError: Expecting: ['quoted_identifier', 'unquoted_identifier', 'lbracket', 'lbrace'], got: number: Parse error at column 6, token "47" (NUMBER), for expression:
"banks.47"

**Pull request merged**